build-system/tasks/check-types.js

```diff
@@ -48,6 +48,9 @@ const TSC_TYPECHECK_TARGETS = {
   'compiler': {
     tsconfig: 'src/compiler/tsconfig.json',
   },
+  'carousel': {
+    tsconfig: 'extensions/amp-carousel/tsconfig.json',
```

Comment:
Hot take: Could `TSC_TYPECHECK_TARGETS` just be a list of directories which contain a `tsconfig.json`?

Reply:
It could be!

Reply:
Done